src/equations/shallow_water_2d.jl

```diff
@@ -837,6 +925,59 @@ end
     return waterheight(u, equations) * pressure(u, equations)
 end
 
+"""
+    calc_wavespeed_roe(u_ll, u_rr, direction::Integer,
+                       equations::ShallowWaterEquations2D)
+
+Calculate Roe-averaged velocity `v_roe` and wavespeed `c_roe = sqrt{g h_roe}` depending on direction.
```

Comment:
See 1D

Reply:
Addressed in ef6d070a63118991967d19ebcfa1fa1a953b41ee